docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,221 @@
+includes:
+
+##### Subscription
+
+We remove subscriptions when they've been ended for over a year, based on their
+`ended_at` value. Once a subscription is ended it [can't currently be made
+active again] and so can be deleted.
+
+##### SubscriberList
+
+We remove lists when they have no active subscriptions and their latest, ended
+subscription is over a year old.
+
+It is possible for a list we deem as historic to be subscribed to again,
+however this seems unlikely due to the list not having any active subscriptions
+for over a year. If this was to happen after the list has been deleted then a
+new `SubscriberList` record will be generated.
+
+##### Subscriber
+
+We remove subscribers once their `created_at` is over a year old and they don't
+have any subscriptions (active or ended).
+
+Subscribers should only be without subscriptions (active or ended) when their
+historic subscriptions have been deleted via the historic deletion worker. This
+means we should be able to delete all subscribers (under a year old too) that
+don't have any subscriptions (active or ended).
+
+However, we should be cautious about doing this so we don't accidently remove
+new subscribers when the application changes. Some examples of where it was
+unintentionally possible to end up with subscribers with no subscriptions can
+be found [here][sub bug] and [here][sub bug two].
+
+### Data which has never been used
+
+We remove subscriber lists which have never had any subscriptions and are over
+seven days old. A scenario of when this can happen is when a user quits their
+signup journey midway through. Seven days was the chosen retention period to
+allow users to complete their signup journey as they have a [seven day limit]
+to confirm their subscription.
+
+As mentioned in the context section, subscriber lists pollute our analytics and
+so we should remove them regularly.
+
+[can't currently be made active again]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-data-retention.md#subscriber-and-subscription
+[seven day limit]: https://github.com/alphagov/email-alert-api/blob/f0c61539f7244bdb52db54d277a46412b728d642/app/services/auth_token_generator_service.rb#L7
+
+## Status
+
+Accepted
+
+## Consequences
+
+### Historic deletion worker
+
+We implemented a [worker] which runs periodically every day at [midday].
+Removing this data was deemed as non urgent and so deleting once a day, in
+hours, seemed sufficient.
+
+### The first historic deletion run
+
+On the first run of the worker which handles deleting historical data we
+removed over 33 million records. A breakdown of the amount of deletions per
+model can be found below:
+
+* ContentChange - 148,617
+
+* MatchedContentChange - 1,825,983
+
+* Message - 15
+
+* MatchedMessage - 16,099
+
+* DigestRun - 677
+
+* DigestRunSubscriber - 30,858,770
+
+* Subscription - 1,046,576
+
+* SubscriberList - 11,470
+
+* Subscriber - 196,376
+
+### Presence of emails during historic deletion
```

Comment:
The presence of emails isn't something new. I think this is really an implementation detail, and a pretty minor one at that. We could combine it with the section about the worker, and maybe comment on how this may have once been a problem, but [ADR 006](https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-006-email-delivery-responsibilities.md) means it's basically impossible now.

Reply:
> I think this is really an implementation detail, and a pretty minor one at that

Hmm true, I've chose not to include it at all given it's deemed as 'minor'.